docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+import io
+import os
+from shlex import split
+from subprocess import PIPE, STDOUT, CalledProcessError, run
+from sys import stdin
+from typing import (Any, List, Optional, Sequence, Type, TypeVar, Union,
+                    overload)
+
+from .exceptions import CommandNotFoundOnImageError
+
+
+class Image:
+    """A class which holds a reference to a given docker image and provides
```

Comment:
In general, docstrings don't need to (and shouldn't) start out like "A class which ..." or "A function to ...". The audience already knows that it's a class or a function based on context, so you can just describe what the class represents or what the function does without preamble.

```suggestion
    """Holds a reference to a given docker image and provides
```